packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -224,26 +257,56 @@ export default function MainFeedLayout({
         emptyScreen: <SearchEmptyScreen />,
       };
     }
-    if (!query.query) {
+
+    if (!config.query) {
       return null;
     }
 
     const getVariables = () => {
-      if (isUpvoted) {
-        return { ...query.variables, period: periods[selectedPeriod].value };
+      if (isUpvoted || feedWithDateRange.includes(tab)) {
+        return { ...config.variables, period: periods[selectedPeriod].value };
+      }
+
+      if (isAnyExplore) {
+        const laptopValue =
+          tab === ExploreTabs.ByDate || isExploreLatest ? 1 : 0;
+        const finalAlgo = isLaptop ? laptopValue : selectedAlgo;
+        return {
+          ...config.variables,
+          ranking: algorithms[finalAlgo].value,
+        };
       }
 
       if (isSortableFeed) {
         return {
-          ...query.variables,
+          ...config.variables,
           ranking: algorithms[selectedAlgo].value,
         };
       }
 
-      return query.variables;
+      return config.variables;
     };
 
     const variables = getVariables();
+    const classes = isAnyExplore && {
+      wrapper: '!min-h-0',
+      emptyScreen: 'mt-8 h-auto',
+    };
```

Comment:
Ah was kind of hoping we could avoid sending this at all.
But if it's too much hassle never mind :)

Reply:
Yeah, we can do that. Less code to maintain. Was just worried the scrollbar and placement of the CTA was quite a turn-off.

https://github.com/dailydotdev/apps/pull/3199#discussion_r1634502509